File: comparison/Python/isort-40000.py
import random

def insert_sort(arr):
    for i in range(1, len(arr)):
        j = i - 1
        ele = arr[i]
        while arr[j] > ele and j >= 0:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = ele
    return arr


def mkRandList(n):
    a=[]
    for i in range(1,n+1):
        a.insert(0, random.randint(0,10000))
    return a

a = mkRandList(40000)
insert_sort(a)
#print(a)

        

File: comparison/OCaml/isort40000.ml
open Format

let rec insert x ylist =
  match ylist with
    [] -> [x]
  | y::ys -> if x<=y
    then x::y::ys
    else y::(insert x ys);;


let rec isort (alist : int list) =
  match alist with
    [] -> []
  | x::xs -> insert x (isort xs);;

  
    
let rec mkRandList n =
  match n with
    0 -> []
  | n -> (Random.int 10000)::(mkRandList (n-1));;


let rec validation alist =
  let rec validation_cons x blist =
    match blist with
      [] -> true
    | y::ys -> if x<=y then (validation_cons y ys) else false
  in
  match alist with
    [] -> true
  | x::xs -> validation_cons x xs;;


let () = print_bool (validation (isort (mkRandList 40000)));;

